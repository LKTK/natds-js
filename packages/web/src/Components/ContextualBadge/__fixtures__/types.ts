import { IContextualBadgeTypes } from "../ContextualBadge.props";

export const types: Record<IContextualBadgeTypes, IContextualBadgeTypes> = {
  standard: "standard",
};

export default types;
