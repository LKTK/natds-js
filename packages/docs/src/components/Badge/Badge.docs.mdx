import { Badge, Provider, themes } from '@naturacosmeticos/natds-web';
import ComponentWithLabel from '../../../addons/ComponentWithLabel';
import { colors, variants, anchors, overlaps } from './sections';

# Badge

---

---

---

---

## Colors

<Provider theme={themes.natura.light}>
  <ComponentWithLabel componentList={colors} />
</Provider>

---

---

## Anchors

<Provider theme={themes.natura.light}>
  <ComponentWithLabel componentList={anchors} />
</Provider>

---

---

## Variants

<Provider theme={themes.natura.light}>
  <ComponentWithLabel componentList={variants} />
</Provider>


## API

```jsx highlight-line="2"
import { Badge } from '@naturacosmeticos/natds-web';

<Badge badgeContent="999" color="primary" variant="dot">
  {children}
</Badge>
```

For more advanced usages, check [Material UI Badge docs](https://material-ui.com/components/badges/)

---

---

## Props

| Name              | Type                                                                                                           |  Default                                    | Description                                                                                                                                     |
| :---------------- | :------------------------------------------------------------------------------------------------------------: | :-----------------------------------------: | :---------------------------------------------------------------------------------------------------------------------------------------------: |
| anchorOrigin      | { horizontal: 'left' \| 'right', vertical: 'bottom' \| 'top' }                                                 | { vertical: 'top', horizontal: 'right' }    | The anchor of the badge.                                                                                                                        |
| badgeContent      | node                                                                                                           |                                             | The content rendered within the badge.                                                                                                          |
| children          | node                                                                                                           |                                             | The badge will be added relative to this node.                                                                                                  |
| classes           |                                                                                                                |                                             | Override or extend the styles applied to the component. See the [CSS API](https://material-ui.com/api/badge/#css) for more details.             |
| color             | 'default' \| 'error' \| 'primary' \| 'secondary'                                                               | 'default'                                   | The color of the component. It supports those theme colors that make sense for this component.                                                  |
| component         | elementType                                                                                                    | 'span'                                      | The component used for the root node. Either a string to use a DOM element or a component.                                                      |
| invisible         | bool                                                                                                           | false                                       | If true, the badge will be invisible.                                                                                                           |
| max               | number                                                                                                         | 99                                          | Max count to show.                                                                                                                              |
| overlap           | 'circle' \| 'rectangle'                                                                                        | 'rectangle'                                 | Wrapped shape the badge should overlap.                                                                                                         |
| showZero          | bool                                                                                                           | false                                       | Controls whether the badge is hidden when badgeContent is zero.                                                                                 |
| variant           | 'dot' \| 'standard'                                                                                            | 'standard'                                  | The variant to use.