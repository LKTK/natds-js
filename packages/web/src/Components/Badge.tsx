import React, { FunctionComponent, forwardRef } from "react";
import { withTheme } from "@material-ui/styles";
import MaterialBadge, { BadgeProps as IBadgeProps } from "@material-ui/core/Badge";

export { BadgeProps as IBadgeProps } from "@material-ui/core/Badge";

export const Badge: FunctionComponent<IBadgeProps> = forwardRef((
  props: IBadgeProps,
  ref: any,
) => <MaterialBadge {...props} ref={ref} />);

export default withTheme(Badge);
