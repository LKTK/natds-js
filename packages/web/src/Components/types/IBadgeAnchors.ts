export interface IBadgeAnchors {
  bottomleft: {
    horizontal: string;
    vertical: string
  };
  topleft: {
    horizontal: string;
    vertical: string
  };
  topright: {
    horizontal: string;
    vertical: string
  };
  bottomright: {
    horizontal: string;
    vertical: string
  }
}
