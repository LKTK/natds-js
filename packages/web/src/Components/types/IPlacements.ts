export interface IPlacements {
  "right-start": string;
  "bottom-start": string;
  top: string;
  left: string;
  bottom: string;
  "left-start": string;
  "top-start": string;
  right: string;
  "bottom-end": string;
  "right-end": string;
  "left-end": string;
  "top-end": string
}
