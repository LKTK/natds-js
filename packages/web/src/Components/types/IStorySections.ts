export type IStorySections = { component: JSX.Element; title: string }[];
