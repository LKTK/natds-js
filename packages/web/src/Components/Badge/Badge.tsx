import * as React from "react";
import MaterialBadge from "@material-ui/core/Badge";
import { IBadgeProps } from "./Badge.props";

/**
 * For more advanced usages, check [Material UI Badge docs](https://material-ui.com/components/badges/)
 *
 * ## Importing
 *
 * ```
 * import { Badge } from '@naturacosmeticos/natds-web';
 * ```
 */
export const Badge = React.forwardRef<unknown, IBadgeProps>((
  props: IBadgeProps,
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  ref: any,
) => <MaterialBadge {...props} ref={ref} />);

Badge.displayName = "Badge";

export default Badge;
