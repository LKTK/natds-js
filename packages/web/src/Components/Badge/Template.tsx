import * as React from "react";

import { Story } from "@storybook/react";
import makeStyles from "@material-ui/core/styles/makeStyles";
import createStyles from "@material-ui/core/styles/createStyles";
import clsx from "clsx";
import { IBadgeProps } from "./Badge.props";
import { IThemeWeb } from "../../Themes";
import Badge from "./Badge";
import * as IconStories from "../Icon/Icon.stories";
import { Icon } from "../Icon";

const useStyles = makeStyles(createStyles((theme: IThemeWeb) => ({
  root: {
    "& > *": {
      marginRight: theme.sizes.small,
    },
  },
  shape: {
    backgroundColor: "#FF00FF1A",

    /**
     * @todo check why IDE shows warning: 'semiX' should probably not be assigned to 'height'
     */
    height: theme.sizes.semiX,
    width: theme.sizes.semiX,
  },
  shapeCircle: {
    borderRadius: theme.shape.badgeBorderRadius,
  },
})), { name: "BadgeDemo" });

export const BadgeShapeCircleContent : Story<IBadgeProps> = (props) => {
  const classes = useStyles();

  // eslint-disable-next-line dot-notation
  return <div className={clsx(classes["shape"], classes["shapeCircle"])} {...props} />;
};

export const BadgeShapeContent : Story<IBadgeProps> = (props) => {
  const classes = useStyles();

  // eslint-disable-next-line dot-notation
  return <div className={classes["shape"]} {...props} />;
};

export const Template : Story<IBadgeProps> = (args) => <Badge {...args}>
  {args.children || args.overlap === "circle" ? <BadgeShapeCircleContent /> : <BadgeShapeContent />}
</Badge>;

export const TemplateWithShapes : Story<IBadgeProps> = (args) => {

  const classes = useStyles();

  // eslint-disable-next-line dot-notation
  return <div className={classes["root"]}>
    <Badge {...args}>
      <BadgeShapeContent />
    </Badge>
    <Badge {...args}>
      <BadgeShapeCircleContent />
    </Badge>
    <Badge {...args}>
      {args.children || <Icon name={"outlined-communication-email"} {...IconStories.OutlinedEmailIcon.args} />}
    </Badge>
  </div>;
};
export default Template;
