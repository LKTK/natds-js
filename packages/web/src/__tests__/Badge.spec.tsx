import * as React from "react";
import { expect } from "chai";
import * as renderer from "react-test-renderer";

import Badge from "../Components/Badge";
import { IBadgeAnchors } from "../Components/types/IBadgeAnchors";

export const anchors: IBadgeAnchors = {
  topleft: {
    horizontal: "left",
    vertical: "top",
  },
  topright: {
    horizontal: "right",
    vertical: "top",
  },
  bottomleft: {
    horizontal: "left",
    vertical: "bottom",
  },
  bottomright: {
    horizontal: "right",
    vertical: "bottom",
  },
};

export const colors: { secondary: string; default: string; error: string; primary: string } = {
  default: "default",
  error: "error",
  primary: "primary",
  secondary: "secondary",
};

export const variants: { standard: string; dot: string } = {
  dot: "dot",
  standard: "standard",
};

export const overlaps: { rectangle: string; circle: string } = {
  circle: "circle",
  rectangle: "rectangle",
};

describe("Badge component", () => {
  Object.keys(colors).forEach((color: string) => {
    test(`should match snapshot - Any valid Badge with ${color} color`, () => {
      const component = renderer.create(<Badge color={color as keyof typeof colors}>Badge</Badge>).toJSON();

      expect(component).matchSnapshot(`Badge snapshot - with ${color} color`);
    });
  });

  Object.values(anchors).forEach((anchor) => {
    test(`should match snapshot - Any valid Badge with ${anchor} anchor`, () => {

      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      const component = renderer.create(<Badge anchorOrigin={anchor as any}>Badge</Badge>).toJSON();

      expect(component).matchSnapshot(`Badge snapshot - with ${anchor} anchor`);
    });
  });

  Object.keys(variants).forEach((variant: string) => {
    test(`should match snapshot - Any valid Badge with ${variant} variant`, () => {
      const component = renderer.create(<Badge variant={variant as keyof typeof variants}>Badge</Badge>).toJSON();

      expect(component).matchSnapshot(`Badge snapshot - with ${variant} variant`);
    });
  });

  Object.keys(overlaps).forEach((overlap: string) => {
    test(`should match snapshot - Any valid Badge with ${overlap} overlap`, () => {
      const component = renderer.create(<Badge overlap={overlap as keyof typeof overlaps}>Badge</Badge>).toJSON();

      expect(component).matchSnapshot(`Badge snapshot - with ${overlap} overlap`);
    });
  });
});
