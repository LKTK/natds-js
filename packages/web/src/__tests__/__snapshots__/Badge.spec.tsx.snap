// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component should match snapshot - Any valid Badge with [object Object] anchor: Badge snapshot - with [object Object] anchor 1`] = `
<span
  className="MuiBadge-root"
>
  Badge
  <span
    className="MuiBadge-badge MuiBadge-anchorOriginTopLeftRectangle MuiBadge-invisible"
  />
</span>
`;

exports[`Badge component should match snapshot - Any valid Badge with [object Object] anchor: Badge snapshot - with [object Object] anchor 2`] = `
<span
  className="MuiBadge-root"
>
  Badge
  <span
    className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-invisible"
  />
</span>
`;

exports[`Badge component should match snapshot - Any valid Badge with [object Object] anchor: Badge snapshot - with [object Object] anchor 3`] = `
<span
  className="MuiBadge-root"
>
  Badge
  <span
    className="MuiBadge-badge MuiBadge-anchorOriginBottomLeftRectangle MuiBadge-invisible"
  />
</span>
`;

exports[`Badge component should match snapshot - Any valid Badge with [object Object] anchor: Badge snapshot - with [object Object] anchor 4`] = `
<span
  className="MuiBadge-root"
>
  Badge
  <span
    className="MuiBadge-badge MuiBadge-anchorOriginBottomRightRectangle MuiBadge-invisible"
  />
</span>
`;

exports[`Badge component should match snapshot - Any valid Badge with circle overlap: Badge snapshot - with circle overlap 1`] = `
<span
  className="MuiBadge-root"
>
  Badge
  <span
    className="MuiBadge-badge MuiBadge-anchorOriginTopRightCircle MuiBadge-invisible"
  />
</span>
`;

exports[`Badge component should match snapshot - Any valid Badge with default color: Badge snapshot - with default color 1`] = `
<span
  className="MuiBadge-root"
>
  Badge
  <span
    className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-invisible"
  />
</span>
`;

exports[`Badge component should match snapshot - Any valid Badge with dot variant: Badge snapshot - with dot variant 1`] = `
<span
  className="MuiBadge-root"
>
  Badge
  <span
    className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-dot"
  >
    
  </span>
</span>
`;

exports[`Badge component should match snapshot - Any valid Badge with error color: Badge snapshot - with error color 1`] = `
<span
  className="MuiBadge-root"
>
  Badge
  <span
    className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorError MuiBadge-invisible"
  />
</span>
`;

exports[`Badge component should match snapshot - Any valid Badge with primary color: Badge snapshot - with primary color 1`] = `
<span
  className="MuiBadge-root"
>
  Badge
  <span
    className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary MuiBadge-invisible"
  />
</span>
`;

exports[`Badge component should match snapshot - Any valid Badge with rectangle overlap: Badge snapshot - with rectangle overlap 1`] = `
<span
  className="MuiBadge-root"
>
  Badge
  <span
    className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-invisible"
  />
</span>
`;

exports[`Badge component should match snapshot - Any valid Badge with secondary color: Badge snapshot - with secondary color 1`] = `
<span
  className="MuiBadge-root"
>
  Badge
  <span
    className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary MuiBadge-invisible"
  />
</span>
`;

exports[`Badge component should match snapshot - Any valid Badge with standard variant: Badge snapshot - with standard variant 1`] = `
<span
  className="MuiBadge-root"
>
  Badge
  <span
    className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-invisible"
  />
</span>
`;
